FirebasePerformance/Sources/FPRClient.m

```diff
@@ -227,6 +253,15 @@ - (void)processAndLogEvent:(FPRMSGPerfMetric *)event {
     return;
   }
 
+  if (self.isFirstTimeProcessAndLogEvent) {
```

Comment:
Use GCD construct dispatch_once() to call this just once the first time an event is sent. You don't have to have a state variable to do this.

https://developer.apple.com/documentation/dispatch/1447169-dispatch_once

We have used this in our code in many other places.